book/src/volume-cloning.md

```diff
@@ -0,0 +1,56 @@
+# Volume Cloning
+
+## Status and Releases
+
+** Status:** Alpha
+
+Compatible with CSI Version | Supported CSI external Provisioner Version | Min k8s Version | Max k8s version
+--|--|--|--|--|--
+v1.0  | v1.3 | v1.15 | -
+
+## Overview
+
+A Clone is defined as a duplicate of an existing Kubernetes Volume.  For more information on cloning in Kubernetes see the concepts doc for [Volume Cloning](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#volume-cloning).
+
+A storage provider that allows volume cloning as a create feature, may choose to implement volume cloning via a
+control-plane CSI RPC call.
+
+## Implementing Volume cloning functionality
+
+To implement volume cloning the CSI driver MUST:
+
+1. Implement checks for `csi.CreateVolumeRequest.VolumeContentSource` in the plugins `CreateVolume` function implementation.
+2. Implement `CLONE_VOLUME` controller capability.
+
+The cloning feature in Kubernetes is implemented by adding a `DataSource` entry of `kind: PersistentVolumeClaim` to the `csi.CreateVolumeRequest`.  This presents the source volume (PersistentVolumeClaim to clone from) as a generic data source.  The source volume must:
+
+1. Be in the same namespace as the destination volume being provisioned
+2. Be in the same storage class as the destination volume being provisioned
+3. Be in a `Bound` state
+4. Be equal or lesser in size than the destination volume being provisioned.
+
+Note that the `external-provisioner` side car currently handles the checks listed above, as well as verifying the specified source can be fetched using the Kubernetes API.
+
+It is the responsibility of the storage plugin to either implement an expansion after clone if a provision request size is greater than the source.
+
+## Deploying volume clone functionality
+
+The Kubernetes CSI development team maintains the [external-provisioner](external-provisioner.md) which is responsible for detecting requests for a PVC DataSource and providing that information to the plugin during provisioning in the call to the plugins `CreateVolume` function.  It's up to the plugin to check the `csi.CreateVolumeRequest` for a `VolumeContentSource` entry in the CreateVolumeRequest object.
+
+There are no additional side-cars or add on components required.
+
+## Enabling Cloning for CSI volumes in Kubernetes
+
+Volume cloning for CSI volumes is an alpha feature (Kubernetes 1.15) and hence must be explicitly enabled via feature gate:
+
+```
+--feature-gates=VolumePVCDataSource=true
+```
+
+## Example implementation
+
+A trivial example implementation can be found in the [csi-hostpath plugin](https://github.com/kubernetes-csi/csi-driver-host-path) in its implementation of `CreateVolume`.
+
+## Examples
```

Comment:
Is this already included in Kubernetes user docs?

Reply:
yes, I'll remove line #56 as well as delete the example yaml files that I had added.

Reply:
still needs to be removed?

Reply:
For real this time!